### PR TITLE
[Bug][Memtable] fix core dump on int128 because not aligned by 16 byte

### LINKED ISSUE #10766: [Bug] Core dump with test_create_table_with_bloom_filter
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

```cpp
*** Aborted at 1657530179 (unix time) try "date -d @1657530179" if you are using GNU date ***
*** SIGSEGV unkown detail explain (@0x0) received by PID 82723 (TID 0x7f4b0f629700) from PID 0; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/common/signal_handler.h:407
 1# 0x000000318AE32920 in /lib64/libc.so.6
 2# doris::vectorized::IAggregateFunctionDataHelper<doris::vectorized::AggregateFunctionSumData<doris::vectorized::Decimal<__int128> >, doris::vectorized::AggregateFunctionSum<doris::vectorized::Decimal<__int128>, doris::vectorized::Decimal<__int128>, doris::vectorized::AggregateFunctionSumData<doris::vectorized::Decimal<__int128> > > >::create(char*) const at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/vec/aggregate_functions/aggregate_function.h:196
 3# doris::MemTable::_insert_one_row_from_block(doris::MemTable::RowInBlock*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/olap/memtable.cpp:188
 4# doris::MemTable::insert(doris::vectorized::Block const*, std::vector<int, std::allocator<int> > const&) in /home/disk2/pxl/dev/baidu/bdg/doris/core/output/be/lib/doris_be
 5# doris::DeltaWriter::write(doris::vectorized::Block const*, std::vector<int, std::allocator<int> > const&) in /home/disk2/pxl/dev/baidu/bdg/doris/core/output/be/lib/doris_be
 6# doris::Status doris::TabletsChannel::add_batch<doris::PTabletWriterAddBlockRequest, doris::PTabletWriterAddBlockResult>(doris::PTabletWriterAddBlockRequest const&, doris::PTabletWriterAddBlockResult*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/runtime/tablets_channel.h:222
 7# doris::Status doris::LoadChannel::add_batch<doris::PTabletWriterAddBlockRequest, doris::PTabletWriterAddBlockResult>(doris::PTabletWriterAddBlockRequest const&, doris::PTabletWriterAddBlockResult*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/runtime/load_channel.h:151
 8# doris::Status doris::LoadChannelMgr::add_batch<doris::PTabletWriterAddBlockRequest, doris::PTabletWriterAddBlockResult>(doris::PTabletWriterAddBlockRequest const&, doris::PTabletWriterAddBlockResult*) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/runtime/load_channel_mgr.h:137
 9# std::_Function_handler<void (), doris::PInternalServiceImpl::_tablet_writer_add_block(google::protobuf::RpcController*, doris::PTabletWriterAddBlockRequest const*, doris::PTabletWriterAddBlockResult*, google::protobuf::Closure*)::$_1>::_M_invoke(std::_Any_data const&) at /home/disk2/pxl/dev/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:291
10# doris::PriorityThreadPool::work_thread(int) at /home/disk2/pxl/dev/baidu/bdg/doris/core/be/src/util/priority_thread_pool.hpp:138
11# execute_native_thread_routine in /home/disk2/pxl/dev/baidu/bdg/doris/core/output/be/lib/doris_be
12# start_thread in /lib64/libpthread.so.0
13# clone in /lib64/libc.so.6


```

### What You Expected?

fix

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

Issue Number: close #10766
only core dump on clang
## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/memtable.cpp [hunk 1]
```diff
@@ -101,19 +101,14 @@ void MemTable::_init_agg_functions(const vectorized::Block* block) {
     }
 
     for (uint32_t cid = _schema->num_key_columns(); cid < _schema->num_columns(); ++cid) {
+        size_t alignment_of_state = _agg_functions[cid]->align_of_data();
+        size_t module = _total_size_of_aggregate_states % alignment_of_state;
+        if (module) {
+            _total_size_of_aggregate_states += alignment_of_state - module;
+        }
```
Comment (dataroaring) [inline be/src/olap/memtable.cpp:108]: DCHECK(alignment_of_state & (alignment_of_state -1)  == 0);
_total_size_of_aggregate_states = (_total_size_of_aggregate_states + alignment_of_state -1 )  & (~(alignment_of_state -1));

Comment (BiteTheDDDDt) [reply be/src/olap/memtable.cpp:108]: alignment_of_state just is `sizeof(FunctionData)`, not necessarily power of 2

**Comment by BiteTheDDDDt** [comment]
yes





------------------ Original ------------------
From: Xinyi Zou ***@***.***&gt;
Date: Mon,Jul 11,2022 11:55 PM
To: apache/doris ***@***.***&gt;
Cc: Pxl ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [apache/doris] [Bug][Memtable] fix core dump on int128 because not aligned by 16 byte (PR #10775)






@xinyiZzz commented on this pull request.


In be/src/runtime/mem_pool.h:
&gt; @@ -104,9 +104,17 @@ class MemPool {     /// of the current chunk. Creates a new chunk if there aren't any chunks     /// with enough capacity.     uint8_t* allocate(int64_t size, Status* rst = nullptr) {+        // TODO: rethink if DEFAULT_ALIGNMENT should be 82, malloc is aligned by 16.
Do you mean DEFAULT_ALIGNMENT changed from 8 to 16

File: be/src/runtime/mem_pool.h [hunk 2]
```diff
@@ -104,9 +104,17 @@ class MemPool {
     /// of the current chunk. Creates a new chunk if there aren't any chunks
     /// with enough capacity.
     uint8_t* allocate(int64_t size, Status* rst = nullptr) {
+        // TODO: rethink if DEFAULT_ALIGNMENT should be 82, malloc is aligned by 16.
```
Comment (HappenLee) [inline be/src/runtime/mem_pool.h:107]: not 82?

**Comment by xinyiZzz** [comment]
> yes
> […](#)

you can change it directly~

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**